Add all-in-one pass pipeline

Menooker opened the pull request.

Menooker commented:
WIP: Depending on https://github.com/intel/graph-compiler/pull/70 and support of openmp in gcc.

ciyongch commented on lib/gc/Transforms/Pipeline.cpp:
```diff
@@ -0,0 +1,166 @@
+  populateVectorPasses(pm);
+  // back-end, arith/math/vector/memref dialects
+  populateBufferizationPasses(pm);
+  // REMOVE this pass after the TensorPasses are added. Currently we add this
```
Can we add "TODO" to highlight the remaining items to better distinguish with other common comments?

ciyongch commented on src/gc-cpu-runner/gc-cpu-runner.cpp:
```diff
@@ -27,7 +27,11 @@
+
 int main(int argc, char **argv) {
+  // keeps GCCPURuntime linked
+  gc_runtime_keep_alive = 0;
```
Do we still need this?

ciyongch commented on test/gc/Transforms/Pipeline/run.mlir:
```diff
@@ -0,0 +1,23 @@
+// RUN: gc-opt %s --gc-cpu-pipeline | gc-cpu-runner -e main -entry-point-result=void | FileCheck %s
```
We might need to add some complex example to cover the default pipeline.

Menooker replied:
@LongshengDu Please help to add some test cases and check if the pipeline works for the current linalgx and onednngraph dialect. Maybe in another PR, or in this one?

kurapov-peter reviewed (approved):
Looks good, could you please clean up all the commented out code before merge?

kurapov-peter commented:
@AndreyPavlenko, please take a look

AndreyPavlenko commented:
I think it would be nice to use the same code style for the entire code. Currently, one part of the code is in the CamelCase, another - in snake_case.

Menooker commented:
> Can we add "TODO" to highlight the remaining items to better distinguish with other common comments?

OK, changed the comments in pipeline to "todo"s.

> Looks good, could you please clean up all the commented out code before merge?

I have changed most of them to "todo"s. They mark the position where not-yet-implemented MLIR passes should be. The developers of that pass can know where to add to the pipeline in the future.
